### PR TITLE
Add Pager component for Next/Previous links

### LINKED ISSUE #6: Next and Previous Page Links
**Is your feature request related to a problem? Please describe.**

It is important that we have links to both the next and previous page in each page footer for two reasons:

1. It makes mobile nav _much_ easier. Instead of needing to open the hamburger menu and figure out where to go next, you just click the link at the bottom of the page. Huge UX improvement + no-one ends up getting lost or anxious.
2. It subtly emphasizes that there is a logical flow to the overall site. This has not been put together haphazardly: there is an actual pedagogical model underlying it and we should make that clear if possible.

**Describe the solution you'd like**

At the bottom of each content page, we should have a `Next` button on the right side of the page and a `Previous` button on the left. These buttons should also display the title of the pages to which they link. See the [current kernel site](https://kernel.community/module-2/better-questions/) for a clear example of what is meant. If possible, the design should be exactly the same. However, if not, we can embed these button in the page itself (as opposed to having them as part of the footer), using the same structure (`Next` + Page Title + arrow) as Kernel, kinda like this:

Screenshot from 2021-03-31 12-40-28

**Notes*

1. This should be an option that can be configured in the frontmatter of each page, if possible. It should default to true (i.e. buttons appear).
2. These buttons need to appear on desktop and mobile. Mobile, especially, is important.
3. It would likely be easiest to use the current [calculateTreeData](https://github.com/kernel-community/kernel-v2/blob/main/src/modules/navigation/calculateTreeData.js) setup.

### PR DESCRIPTION
**Pull request by harrisrobin**
### Attention (CC)
@andytudhope 

## Summary of changes

This PR adds a <Pager /> component which does the following:
1. renders Next and Previous links under the article
2. figures out the next/previous links based on the sideNavData generated by `calculateTreeData`
3. handles edge cases such as no previous section or previous section being nested or not.

This should not impact any other functionality or break anything else as it's pretty isolated and no new dependencies were added.

This PR also removes .vscode and adds it to .gitignore (otherwise my personal config would be committed) 

## Relevant Context

- Feature: This PR closes #6 

## How Has This Been Tested?

I tested it by navigating the Learn and Play sections of the site entirely with the Pager's "next" and "previous" buttons.

## Screenshots (if appropriate):
quickdemo

--- 
## Types of changes

- [ ] Content edit, File upload, content addition
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Technical Documentation Update
- [ ] Other (please describe):

## Does this introduce a breaking change?

_Breaking change is defined as_ **"A fix or feature that would cause existing functionality to change."**

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added/changed necessary documentation (if appropriate)
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/modules/layouts/global.css [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-  -webkit-font-smoothing: subpixel-antialiased;
-  -moz-osx-font-smoothing: grayscale;
   backface-visibility: hidden;
 }
 
+* {
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
+
```
Comment (harrisrobin) [inline src/modules/layouts/global.css:10]: This actually drastically improved font rendering, at least for me on mac OS.

**Comment by andytudhope** [comment]
First of all @harrisrobin - thank you so very, very much. This is genuinely impressive work for your first contribution. I love the animations and attention to detail.

Some notes, before @MaximumCrash does his own review here.

1. This breaks the home page:

Screenshot from 2021-04-19 13-10-33

2. It does not work for top-level pages, i.e. when you click on "Learn" or "Start" or "Question" etc. no buttons appear on those pages (which is my bad for not specifying that as a requirement if possible).
3. If there are no nested pages like, for example, in "Question" or "Resources", then no buttons appear (though you could derive the order of those pages from calculateTreeData afaik).

**Comment by andytudhope** [comment]
One other relevant comment from Harris, @MaximumCrash 

"it was not feasible to make it look like the previous design without changing the layout. The `default_layout` is actually nested in a width constrained div which means that I could not go out of bounds and do the full-width solution you have in Kernel v1 without having to change the layout and hierarchy of some components. Since I wanted to keep this simple, I simply placed it under <Article/> which worked out nicely imo and it does take the full-width on mobile."

**Review by andytudhope — APPROVED** [review]
I'm really happy with this, thank you so very much @harrisrobin :heart: 

Will wait for @MaximumCrash before merging.

**Comment by MaximumCrash** [comment]
Firstly kudos to @harrisrobin for being able to add into my ballistic `calculateTreeData` algorithm. It's good to know it's somewhat easy to follow 😅 

TLDR: I feel like there's a good enough amount of work to close this PR out and get started on 

My only reservations is in the use of the @emotion/styled since it breaks away from the existing pattern of using `sx` styles on html/react components. However it's opened the question of whether theme-ui is still the best option for this system. For context, there was a flashing styles issue that I discovered last summer during the MakerDAO portal project that didn't make too much sense to fix, until I looked deeper into how emotion, and similar css-in-js libraries like it, work. Because emotion applies something like this:
```
<css> ... </css>
```

interspersed into dom elements there's a massive performance issue with loading styles. Too many paints and all. The more styles, the worse this gets. 

My gut reaction is to consider swapping in something like tailwindcss, but given the nature of the portal's philosophy I'm wavering on whether to stick with theme-ui, and be default emotion. 

> I'm not sure why this happens. I haven't really done anything in my code to cause this and this behavior seems to come from
> "@modules/navigation/Link" so perhaps @MaximumCrash might know why this happens better than me, though I haven't really looked into that component yet.

For this, the Link component is over engineered to stop situations where any writer in the old portal would write bad actor links. There was also a number of specific logic checks based on the previous designers needs so that can change. But I can attack this today immediately.

**Comment by andytudhope** [comment]
I love both of your work, thank you very much :heart:
